### ISSUE TITLE
Grid.sort don't works on StartUp

### ISSUE THREAD

**Issue by abierlerxdev** [vaadin/vaadin-grid-flow#465]
Grids don't get sorted on startup, only if code is executed **after** the grid is loaded.

### How to reproduce:
```
@Route("")
public class ComboBoxDemo3 extends Div
{
	
	private final Grid<LocalDate> grDate = new Grid<>();
	
	public ComboBoxDemo3()
	{
		
		final Column<LocalDate> col = this.grDate.addColumn(LocalDate::getDayOfMonth).setHeader("Weekday");

		this.grDate.setItems(Arrays.asList(
			LocalDate.of(2018,1,5),
			LocalDate.of(2018,1,1),
			LocalDate.of(2018,1,3),
			LocalDate.of(2018,1,2)
			));

		this.grDate.sort(new GridSortOrderBuilder<LocalDate>().thenDesc(col).build());
		
		this.add(this.grDate);
		
		this.add(new Button("Test",l -> {
			this.grDate.setItems(Arrays.asList(
				LocalDate.of(2018,1,5),
				LocalDate.of(2018,1,1),
				LocalDate.of(2018,1,3),
				LocalDate.of(2018,1,2)
				));
			this.grDate.sort(new GridSortOrderBuilder<LocalDate>().thenDesc(col).build());
		}));
	}
}
```
On startup the grid isn't sorted, but it should. If you press the button, the grid get's sorted.

### Example:
On StartUp:
![grafik](https://user-images.githubusercontent.com/45384811/50093677-a8ba6c00-0211-11e9-9113-3e36adefac11.png)

After clicking the button:
![grafik](https://user-images.githubusercontent.com/45384811/50093619-80327200-0211-11e9-81ff-8f176d730028.png)

**Comment by Springrbua** [comment]
I am having the same problem and I was able to locate the possible issue.
I registered a `SortListener` and noticed, that it gets notified twice:
1. SortEvent coming from the server (`isFromClient() == false`) with the correct sort.
2. SortEvent coming from the client (`isFromClient() == true`) with empty sort.

I tryed to debug it and I found, that `sortChanged` is fired when some properties of the grid are modified. This modifications causes a call-chain which finally results in a `sortChangeEvent`.
The first `sortChange` is caused by the `updateSize`-Function of the `gridConntector`.
It seems to fire the event, because the `_previousSorters` changed from `undefined` to `[]`.
Here some screenshots from the developer-tools:

**Call Stack:**
![image](https://user-images.githubusercontent.com/18258317/51184481-aad91e80-18d4-11e9-8f37-b058bb0c9b1b.png)

**_propertiesChanged:**
![image](https://user-images.githubusercontent.com/18258317/51184521-ca704700-18d4-11e9-9b3c-cea285393294.png)

![image](https://user-images.githubusercontent.com/18258317/51184546-da882680-18d4-11e9-9c81-b1203a474245.png)


![image](https://user-images.githubusercontent.com/18258317/51184559-e673e880-18d4-11e9-9e61-dab185dc58ba.png)

My code:
```
private Grid<Model> grid = new Grid<>();

public MainView() {
  grid.setSizeFull();

  grid.addColumn(Model::getId).setHeader("ID").setSortable(true);
  Column<Model> col = grid.addColumn(Model::getName).setHeader("Name").setSortable(true);

  add(grid);
  setHeight("100vh");
  updateList();

  ArrayList<GridSortOrder<Model>> sort = new ArrayList<>();
  sort.add(new GridSortOrder<>(col, SortDirection.ASCENDING));

  grid.addSortListener(event -> {
  if (event.isFromClient()) {
    System.out.println("Client sort: " + event.getSortOrder().size());
	// grid.sort(sort);    // <----- Remove comment
  } else
    System.out.println("Server sort: " + event.getSortOrder().size());
  });

  grid.sort(sort);

  grid.addItemClickListener(event -> {
  grid.sort(sort);
  });
}

private void updateList() {
  grid.setItems(new Model(1, "B"), new Model(2, "X"), new Model(3, "A"), new Model(4, "C"), new Model(5, "F"),
    new Model(6, "P"));
}

private static class Model {

  private final int id;
  private final String name;

  public Model(int id, String name) {
  this.id = id;
  this.name = name;
  }

  public int getId() {
  return id;
  }

  public String getName() {
  return name;
  }
}
```

With this code I get the following output:
```
Server sort: 1
Client sort: 0
```

The `Server sort` is caused by `grid.sort(sort);` directly after registering the listener.
The `Cleint sort` is caused by `grid$connector.updateSize`.  
If I add a `grid.sort(sort)` inside the `SortListener`, I get the following output:
```
Server sort: 1
Client sort: 0
Server sort: 1
Client sort: 0
Server sort: 1
Client sort: 0
Server sort: 1
```
The first two outputs are caused by the same events as before.
The other `Client sort`s are caused by the `connectedCallback` in the [`vaadin-grid-sorter.html`](https://github.com/vaadin/vaadin-grid/blob/master/src/vaadin-grid-sorter.html#L156), which sets the `_isConnected`-property and causes a `sortChange`-Event.

**Comment by T3rm1** [comment]
Still not working even after 1 month.

**Comment by ggecy** [comment]
I have the same problem. This makes it impossible to set the initial sorting on grid.

**Comment by Springrbua** [comment]
I just tested it with Vaadin 13, but the issue persists.
Is there some kind of work-around for this problem?

**Comment by ggecy** [comment]
The issue is still there as far as I am aware, however I have a workaround. The point is to sort the grid after the initial data load, grid fires an event 'loading-changed' when loading data. I have created a custom subclass where you can add a listener and if the event.isLoading() is false, you can sort the grid and it will work:

```
public class LoadingAwareGrid<T> extends Grid<T> {

    public LoadingAwareGrid() {
    }

    public LoadingAwareGrid(int pageSize) {
        super(pageSize);
    }

    public LoadingAwareGrid(Class<T> beanType, boolean autoCreateColumns) {
        super(beanType, autoCreateColumns);
    }

    public LoadingAwareGrid(Class<T> beanType) {
        super(beanType);
    }

    public Registration addLoadingListener(
            ComponentEventListener<LoadingChangedEvent> listener) {
        return ComponentUtil.addListener(this, LoadingChangedEvent.class, listener);
    }

    @DomEvent("loading-changed")
    public static class LoadingChangedEvent extends ComponentEvent<LoadingAwareGrid> {

        private boolean loading;

        public LoadingChangedEvent(LoadingAwareGrid source, boolean fromClient,
                                   @EventData("element.loading") boolean loading) {
            super(source, fromClient);
            this.loading = loading;
        }

        public boolean isLoading() {
            return loading;
        }
    }
}
```

**Comment by JohannesLaidler** [comment]
Any estimates, when this will get fixed? This is a major bug. Some grids have to be sorted on startup in a certain way to be useful for the user.

**Comment by cedricaebi** [comment]
@JohannesLaidler I tried @ggecy‘s approach. But it didn‘t work for me. It sorted initially. But everytime I tried to sort the grid after the initial loading to a different sort order it fired the load-change event and overwrote my sorting. So yep. I need a solution from vaadin

**Comment by ggecy** [comment]
> @JohannesLaidler I tried @ggecy‘s approach. But it didn‘t work for me. It sorted initially. But everytime I tried to sort the grid after the initial loading to a different sort order it fired the load-change event and overwrote my sorting. So yep. I need a solution from vaadin

Well, you can remove the listener after initial sort or add some condition so it only runs once. The disadvantage of this approach (and why I am also waiting for fix) is that the grid is loading the data twice.

**Comment by pekam** [comment]
Looks like this is a duplicate of #13, for which there is already a fix pending: #672.
